lib/libzpool/util.c

```diff
@@ -300,3 +304,316 @@ set_global_var(char *arg)
 
 	return (0);
 }
+
+static void
+locate_binary_path(const char *binary, char *path, size_t len)
+{
+	/*
+	 * Try to use the in-tree path. If not successful, just
+	 * allow execv to search for binary through PATH.
+	 */
+	if (realpath(getexecname(), path) == NULL)
+		goto filename;
+
+	/*
+	 * our executable basename is typically "ztest" or "zdb"
+	 */
+	char *basename = strrchr(path, '/');
+	if (basename == NULL)
+		goto filename;
+	basename++;
+
+	/*
+	 * in-tree match is typically for "/cmd/ztest/" or "/cmd/zdb/"
+	 */
+	char match[MAXNAMELEN];
+	(void) snprintf(match, sizeof (match), "/cmd/%s/", basename);
+
+	char *parent = strstr(path, match);
+	if (parent != NULL) {
+		struct stat s;
+		size_t resid = len - (parent - path);
+		(void) snprintf(parent, resid, "/cmd/%s/%s", binary, binary);
+		if (stat(path, &s) == 0)
+			return;
+	}
+
+filename:
+	strlcpy(path, binary, len);
+}
+
+static int
+run_zpool_import_config(const char *pool, const char *config, int nsearch,
+    char *searchdirs[])
+{
+	char *bin;
+	char **argv;
+	const int len = MAXPATHLEN + MAXNAMELEN + 20;
+	int result;
+	int argc = 0, argsiz;
+
+	bin = umem_alloc(len, UMEM_NOFAIL);
+	locate_binary_path("zpool", bin, len);
+
+	/*
+	 * basic argv set = {"zpool", "import", "-C", tmpfile, pool, NULL}
+	 */
+	argsiz = 6 * sizeof (char *);
+	if (nsearch > 0)
+		argsiz += (nsearch * 2) * sizeof (char *);
+	argv = umem_alloc(argsiz, UMEM_NOFAIL);
+
+	argv[argc++] = bin;
+	argv[argc++] = "import";
+	argv[argc++] = "-C";
+	argv[argc++] = (char *)config;
+	for (int i = 0; i < nsearch; i++) {
+		argv[argc++] = "-d";
+		argv[argc++] = searchdirs[i];
+	}
+	argv[argc++] = (char *)pool;
+	argv[argc++] = NULL;
+
+	pid_t pid = vfork();
+	switch (pid) {
+	case -1:
+		_exit(-1);
+
+	case 0: /* Child process */
+	{
+		int fd = open("/dev/null", O_WRONLY);
+
+		if (fd > 0) {
+			(void) dup2(fd, 1);
+			(void) dup2(fd, 2);
+			(void) close(fd);
+		}
+
+		if (bin[0] == '/')
+			(void) execvp(bin, argv);
+		else
+			(void) execv(bin, argv);
```

Comment:
I think you may have `execvp` (search $PATH) and `execv` switched.  But execvp might work for both cases:
```
       The  execlp(),  execvp(),  and  execvpe()  functions  duplicate the actions of the shell in
       searching for an executable file if the specified filename does not  contain  a  slash  (/)
       character.  The file is sought in the colon-separated list of directory pathnames specified
       in the PATH environment variable.  If this variable isn't defined, the path  list  defaults
       to  a  list  that  includes  the directories returned by confstr(_CS_PATH) (which typically
       returns the value "/bin:/usr/bin") and possibly also the  current  working  directory;  see
       NOTES for further details.

       If the specified filename includes a slash character, then PATH is ignored, and the file at
       the specified pathname is executed.```